src/test/java/com/scalar/cassy/transferer/AzureBlobFileDownloaderTest.java

```diff
@@ -40,64 +42,57 @@
   private static final String ANY_TARGET_IP = "target_ip";
   private static final String ANY_TMP_DATA_DIR = "tmp_data_dir";
   private static final String ANY_STOREBASE_URI = "container_name";
-  @Mock private BlobContainerAsyncClient containerClient;
-  @Mock private BlobAsyncClient blobClient;
-  private AzureBlobFileDownloader downloader;
-  @Mock private PagedFlux<BlobItem> pagedFlux;
+  @Mock private BlobContainerClient containerClient;
+  @Mock private BlobClient blobClient;
+  @Mock private OutputStream outputStream;
+  @Spy @InjectMocks private AzureBlobFileDownloader downloader;
 
   @Before
   public void setUp() {
     MockitoAnnotations.initMocks(this);
-    downloader = new AzureBlobFileDownloader(this.containerClient);
   }
 
   @Test
   public void download_TwoFiles_ShouldDownloadBothFiles() throws Exception {
     // Arrange
     RestoreConfig config =
         new RestoreConfig(getProperties(BackupType.NODE_SNAPSHOT, ANY_TMP_DATA_DIR));
-    when(containerClient.listBlobs(any())).thenReturn(pagedFlux);
     File file1 = createTempFile("file_1", "Lorem");
     File file2 = createTempFile("file_2", "Lorem ipsum");
     Assertions.assertThat(file2.delete()).isTrue();
-    when(pagedFlux.toIterable()).thenReturn(getBlobItems(file1, file2));
-    when(containerClient.getBlobAsyncClient(file1.getName())).thenReturn(blobClient);
-    when(blobClient.downloadToFile(anyString())).thenReturn(Mono.just(mock(BlobProperties.class)));
-    when(containerClient.getBlobAsyncClient(file2.getName())).thenReturn(blobClient);
-    when(blobClient.downloadToFile(anyString())).thenReturn(Mono.just(mock(BlobProperties.class)));
+    doReturn(getBlobItems(file1, file2)).when(downloader).listBlobs(any());
```

Comment:
Sorry I don't get why we need to have `listBlobs` method with @VisibleForTesting.
Can we do something like this simply?
```
when(containerClient.listBlobs(any(ListBlobsOptions.class), any(Duration.class))).thenReturn(getBlobItems(file1, file2));
```
```

Reply:
Sorry, I tried this before but I couldn't resolve this because listBlobs returns `PagedIterable<BlobItem>`. I couldn't create a `PagedIterable<BlobItem>` object for `getBlobItems(File... files)`. 

<img width="857" alt="Screen Shot 2021-06-08 at 11 29 24 AM" src="https://user-images.githubusercontent.com/58680313/121113333-d44fc780-c84c-11eb-9521-d1049a84031b.png">

So I resolved the issue using the listBlobs method with @VisibleForTesting.

Reply:
That's because getBlobItems returns Iterable.
Can it return PagedIterable?

Reply:
Yesterday I tried to create a PagedIterable object but I couldn't resolve it.
And I tried type cast also.
I will try once again.

Reply:
If you can't do it easily, we can go with the current solution.

Reply:
@feeblefakie san,
> If you can't do it easily, we can go with the current solution.

I hope we can continue with this solution. 
Sorry, I tried to resolve it with the PagedIterable object but I couldn't